docs/classes.rst

```diff
@@ -185,10 +185,9 @@ inheritance relationship:
         std::string bark() const { return "woof!"; }
     };
 
-There are three different ways of indicating a hierarchical relationship to
+There are two different ways of indicating a hierarchical relationship to
```

Comment:
This deprecation commit is probably worth cherry-picking into master.  It doesn't make a whole lot of sense to have three different methods for a base class anyway, I just thought it would be presumptuous to deprecated it in my PR.